### PR TITLE
docs: Synthetics monitoring

### PR DESCRIPTION
**Pull request by bmorelli25**
## Summary

Synthetic monitoring documentation! At this point, I think it makes sense to keep Uptime docs separate from the Synthetics docs. There are a couple of reasons:

* One is still in alpha and it easily separates the two
* At least with the alpha, Uptime is easier to get started with. I don't think we should clutter the getting started experience of either by introducing them at the same time.

## Preview

A preview of this documentation is available: https://observability-docs_209.docs-preview.app.elstc.co/guide/en/observability/master/synthetics-monitoring.html

## Related issues

Closes: https://github.com/elastic/observability-docs/issues/168
Closes: https://github.com/elastic/observability-docs/issues/206

### CODE REVIEW

**Comment by andrewvc** [comment]
These docs are great, I haven't had a chance to actually test them carefully, but on my initial read-through these are just great. Thanks so much!

File: docs/en/observability/synthetics-command-reference.asciidoc [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+= Synthetics command reference
+
+++++
+<titleabbrev>Command reference</titleabbrev>
+++++
+
+experimental::[]
+
+[discrete]
+== `npx @elastic/synthetics` command
```
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-command-reference.asciidoc:11]: this would be become as we cannot have scoped CLI bin names`npx synthetics`

Comment (bmorelli25) [reply docs/en/observability/synthetics-command-reference.asciidoc:11]: I'm not sure I understand

File: docs/en/observability/synthetics-get-started.asciidoc [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  schedule: "@every 1m"
+  script: |- <2>
+    step("load homepage", async () => {
+        await page.goto('https://www.elastic.co');
+    });
+    step("hover over products menu", async () => {
+        await page.hover('css=[data-nav-item=products]');
+    });
+----
+<1> Each `monitor` gets its own ID in the {uptime-app} and therefor its own schedule entry.
```
Comment (graphaelli) [inline docs/en/observability/synthetics-get-started.asciidoc:71]:
```suggestion
<1> Each `monitor` gets its own ID in the {uptime-app} and therefore its own schedule entry.
```

Comment (bmorelli25) [reply docs/en/observability/synthetics-get-started.asciidoc:71]: 🙇

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by bmorelli25** [comment]
This PR is (finally) ready for review. I'll likely merge it early tomorrow morning and open a follow-up PR with any remaining reviewer suggestions.

File: docs/en/observability/synthetics-create-test.asciidoc [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+    @keyup.enter="addTodo">
+</header>
+----
+
+You can verify that `new-todo` class element contains the expected `placeholder` (input hint)
+with the following test:
+
+[source,js]
+----
+const input = await page.$('input.new-todo'); <1>
```
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-create-test.asciidoc:109]: should we have this inside step?

File: docs/en/observability/synthetics-create-test.asciidoc [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+
+[discrete]
+[[synthetics-sample-test]]
+=== Sample Synthetic test
+
+A complete example of a basic synthetic test looks like this:
+
+[source,js]
+----
+import { journey, step } from '@elastic/synthetics';
```
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-create-test.asciidoc:127]: add `import { deepStrictEqual } from 'assert'; `

File: docs/en/observability/synthetics-create-test.asciidoc [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+// Should we doc required technologies?
+// Node.js, npx, typescript, etc.?
+
+As explained in the <<synthetics-quickstart,quickstart>>, there are two ways to run synthetic tests:
+
+* Run as an inline journey -- copy/paste into `heartbeat.yml`
+* Run a test suite -- import your tests into Heartbeat
+
+Which option is right for you? That depends.
+If you only have a few tests, an inline journey might be the easiest workflow --
```
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-create-test.asciidoc:154]:
```suggestion
If you want to invoke a single journey with few steps, an inline journey might be the easiest workflow --
```

File: docs/en/observability/synthetics-create-test.asciidoc [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+
+[discrete]
+[[synthetics-inline-journey]]
+=== Run an inline journey
+
+The easiest way to run a synthetic test is by creating an inline journey.
+The `journey` keyword isn't required, and access to variables like `page` and `params` is automatic.
+
+Copy and paste your test steps into `heartbeat.yml`.
+Heartbeat will fork Node.js, schedule your tests, and start headless chrome.
```
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-create-test.asciidoc:167]:
```suggestion
Heartbeat spawns a separate Node.js process, schedule your tests, and runs it on chromium browser.
```

File: docs/en/observability/synthetics-create-test.asciidoc [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+step("load homepage", async () => {
+    await page.goto('https://www.elastic.co');
+});
+step("hover over products menu", async () => {
+    await page.hover('css=[data-nav-item=products]');
+});
+----
+
+To run this, or any other inline example locally, change into the directory of your test,
+and pipe the file contents to the `elastic-synthetics` command.
```
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-create-test.asciidoc:185]:
```suggestion
and pipe the file contents to the `npx @elastic/synthetics` command.
```

File: docs/en/observability/synthetics-create-test.asciidoc [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+
+For example:
+
+[source,sh]
+----
+cat examples/inline/short.js | npx synthetics --inline
```
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-create-test.asciidoc:191]:
```suggestion
cat examples/inline/short.js | npx @elastic/synthetics --inline
```

File: docs/en/observability/synthetics-create-test.asciidoc [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+[discrete]
+[[synthetics-test-suite]]
+=== Run a test suite
+
+If you have a suite of tests you'd like to implement, you can use Elastic synthetics as a library.
+In this method, you use Docker to run both Heartbeat and `elastic-synthetics`.
+
+// [discrete]
+// [[synthetics-suite-install]]
+// ==== Step 1: Install `elastic-synthetics`
```
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-create-test.asciidoc:235]:
```suggestion
// ==== Step 1: Install `@elastic/synthetics`
```

File: docs/en/observability/synthetics-create-test.asciidoc [hunk 10]
```diff
@@ -0,0 +1,310 @@
...
+
+If you haven't already, clone the https://github.com/elastic/synthetics[elastic/synthetics] repository
```
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-create-test.asciidoc:237]: We published the package, so this shouldnt be necessary. Users can do `npm install -g @elastic/synthetics`

File: docs/en/observability/synthetics-create-test.asciidoc [hunk 11]
```diff
@@ -0,0 +1,310 @@
...
+----
+cd examples/todos/ && npm install
+----
+
+From the root of the `synthetics` repo, you can now run the provided tests.
+Any files matching the filename `*.journey.*` will be run.
+
+[source,sh]
+----
+npx elastic-synthetics examples/todos
```
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-create-test.asciidoc:271]:
```suggestion
npx @elastic/synthetics examples/todos
```

File: docs/en/observability/synthetics-create-test.asciidoc [hunk 12]
```diff
@@ -0,0 +1,310 @@
...
+// [discrete]
+// [[synthetics-suite-create]]
+// ==== Step 2: Create your tests
+
+Now it's time to write your tests:
+
+. Create a new Node.js project.
+. Create a `javascript` or `typescript` file that imports your tests.
+All synthetic test files must use the `.journey.ts` or `.journey.js` file extension.
+. Call `run`, which is imported from `elastic-synthetics` and runs the CLI application.
```
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-create-test.asciidoc:255]: Don't need this step one when invoked via CLI.

File: docs/en/observability/synthetics-monitoring.asciidoc [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+[[how-synthetics-works]]
+== How does it work?
+
+While there are different ways to utilize Elastic's synthetic monitoring,
+the easiest is with our provided Docker image:
+
+// Operational use case screenshot
+image::images/synthetics-overview.png[Synthetics overview]
+
+In this use-case, Docker spins up Heartbeat and the `@elastic/synthetics` package.
```
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-monitoring.asciidoc:39]:
```suggestion
In this use-case, Docker spins up Heartbeat which has `@elastic/synthetics` package installed.
```

File: docs/en/observability/synthetics-monitoring.asciidoc [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+Heartbeat handles the scheduling of test runs and supervises the execution of the
+`@elastic/synthetics` package.
+`@elasic/synthetics` runs your tests by invoking the Playwright library and starting a new
+browser process.
+Performance data is then gathered and reported back to Heartbeat,
+where it is formatted and sent to the Elastic Stack.
+At this point, the data is available in the Uptime app to view, analyze, and alert on.
+
+// REVIEWERS
+// Do we need a separate image for the --> suite of tests in user-controlled git repo?
```
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-monitoring.asciidoc:49]: I dont think at this point.

**Review by vigneshshanmugam — COMMENTED** [review]
Great work @bmorelli25, Added some suggestions.

File: docs/en/observability/observability-introduction.asciidoc [hunk 15]
```diff
@@ -54,6 +54,21 @@ uptime, web site endpoints, and API monitoring. For detailed
...
+[float]
+[[user-experience-overview]]
+== User experience data
+
+User Experience data, powered by the APM Real User Monitoring (RUM) agent,
+provides a way to quantify and analyze the perceived performance of your web application.
+
+[float]
+[[synthetic-monitoring-overview]]
+== Synthetic monitoring data (Alpha)
```
[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/en/observability/observability-introduction.asciidoc:66]: Indeed we are -- good catch

File: docs/en/observability/synthetics-create-test.asciidoc (hunk 3 above)
Comment (bmorelli25) [reply docs/en/observability/synthetics-create-test.asciidoc:109]: After looking at this again, yes, I think you're right.

**Comment by bmorelli25** [comment]
@vigneshshanmugam -- Fantastic review. Thanks so much!

File: docs/en/observability/synthetics-command-reference.asciidoc [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+
+Have a question? Want to leave feedback? Visit the
+https://discuss.elastic.co/tags/c/observability/uptime/75/synthetics[Synthetics discussion forum].
+====
+
+[discrete]
+[[elastic-synthetics-command]]
+== `elastic-synthetics`
+
+Heartbeat uses the `elastic-synthetics` command to run and report synthetic tests.
```
[the remark this answers is not in the document]
Comment (andrewvc) [reply docs/en/observability/synthetics-command-reference.asciidoc:22]: Yes, +1 on making this the consistent choice

File: docs/en/observability/synthetics-monitoring.asciidoc [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+This functionality is experimental and may be changed or removed completely in a future release.
+Elastic will take a best effort approach to fix any issues, but experimental features are not subject
+to the support SLA of official GA features.
+
+Have a question? Want to leave feedback? Visit the
+https://discuss.elastic.co/tags/c/observability/uptime/75/synthetics[Synthetics discussion forum].
+====
+
+[role="screenshot"]
+image::images/synthetic-app-overview.png[Synthetics app overview]
```
Comment (andrewvc) [inline docs/en/observability/synthetics-monitoring.asciidoc:19]: This image was taken when the `url` links were broken. Could we swap it with a more recent capture that should have that fixed?

File: docs/en/observability/synthetics-monitoring.asciidoc [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+While there are different ways to utilize Elastic's synthetic monitoring,
+the easiest is with our provided Docker image:
+
+// Operational use case screenshot
+image::images/synthetics-overview.png[Synthetics overview]
+
+In this use-case, Docker spins up Heartbeat, which has the `@elastic/synthetics` package installed.
+Heartbeat handles the scheduling of test runs and supervises the execution of the
+`@elastic/synthetics` package.
+`@elasic/synthetics` runs your tests by invoking the Playwright library and starting a new
```
Comment (andrewvc) [inline docs/en/observability/synthetics-monitoring.asciidoc:42]: `s/elasic/elastic/`

File: docs/en/observability/synthetics-quickstart.asciidoc [hunk 19]
```diff
@@ -0,0 +1,180 @@
+[[synthetics-quickstart]]
+= Quickstart: Synthetics monitoring via Docker
```
Comment (andrewvc) [inline docs/en/observability/synthetics-quickstart.asciidoc:2]: `s/synthetics/synthetic/`

File: docs/en/observability/synthetics-monitoring.asciidoc [hunk 20]
```diff
@@ -0,0 +1,46 @@
+[[synthetics-monitoring]]
+= Synthetics monitoring
```
Comment (andrewvc) [inline docs/en/observability/synthetics-monitoring.asciidoc:2]: s/`synthetics/synthetic/`

File: docs/en/observability/synthetics-monitoring.asciidoc [hunk 21]
```diff
@@ -0,0 +1,46 @@
...
+
+++++
+<titleabbrev>Synthetics monitoring</titleabbrev>
```
Comment (andrewvc) [inline docs/en/observability/synthetics-monitoring.asciidoc:5]: `s/synthetics/synthetic/`

**Review by andrewvc — APPROVED** [review]
LGTM after some minor flaws are fixed.

File: docs/en/observability/synthetics-monitoring.asciidoc (hunk 17 above)
Comment (bmorelli25) [reply docs/en/observability/synthetics-monitoring.asciidoc]: Done.

**Comment by bmorelli25** [comment]
I'm going to merge this. Please feel free to leave additional feedback/comments and I'll address them in a new PR. Thanks for all the help!

**Pull request merged**